src/raftstore/store/store.rs

```diff
@@ -930,8 +937,16 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         if !wb.is_empty() {
             let mut write_opts = WriteOptions::new();
             write_opts.set_sync(self.cfg.sync_log);
-            self.engine.write_opt(wb, &write_opts).unwrap_or_else(|e| {
-                panic!("{} failed to save append result: {:?}", self.tag, e);
+            self.raft_engine.write_opt(wb, &write_opts).unwrap_or_else(|e| {
+                panic!("{} failed to save raft append result: {:?}", self.tag, e);
+            });
+        }
+
+        if !kv_wb.is_empty() {
+            let mut write_opts = WriteOptions::new();
```

Comment:
we can only enter here when we apply a snapshot. How about we update the applying state in the raft rocksdb before but fail here? Is it a corner case?